src/mpi/comm/comm_impl.c

```diff
@@ -601,7 +601,10 @@ static int create_peer_comm(MPIR_Lpid my_lpid, MPIR_Lpid remote_lpid,
 
     bool is_low_group = (lpid_cmp(my_lpid, remote_lpid) < 0);
 
-    MPIR_Lpid peer_map[2];
+    MPIR_Lpid *peer_map;
+    peer_map = MPL_malloc(2 * sizeof(MPIR_Lpid), MPL_MEM_OTHER);
```

Comment:
is there a corresponding free?

Reply:
The map is consumed by `MPIR_Group_create_map` on line 637. It is a bit obscure, but I was trying to avoid duplicating a potentially large proc map array.